src/EnergyPlus/HeatingCoils.cc

```diff
@@ -1274,6 +1258,31 @@ namespace HeatingCoils {
                         state.dataHeatingCoils->ValidSourceType(CoilNum) = true;
                     }
                 }
+            } else if (UtilityRoutines::SameString(Alphas(5), "Coil:Cooling:DX")) {
+                HeatingCoil(CoilNum).ReclaimHeatingSource = HeatObjTypes::COIL_COOLING_DX_NEW;
+                HeatingCoil(CoilNum).ReclaimHeatingSourceIndexNum = CoilCoolingDX::factory(state, Alphas(6));
+                if (HeatingCoil(CoilNum).ReclaimHeatingSourceIndexNum < 0) {
+                    ShowSevereError(
+                        state,
+                        format(
+                            "{}={}, could not find desuperheater coil {}={}", CurrentModuleObject, HeatingCoil(CoilNum).Name, Alphas(5), Alphas(6)));
+                    state.dataHeatingCoils->InputErrorsFound = true;
+                }
+                DataHeatBalance::HeatReclaimDataBase &HeatReclaim =
+                    state.dataCoilCooingDX->coilCoolingDXs[HeatingCoil(CoilNum).ReclaimHeatingSourceIndexNum].reclaimHeat;
```

Comment:
This line should be in an else from the if above to make sure HeatingCoil(CoilNum).ReclaimHeatingSourceIndexNum is >= 0

Reply:
>  rraustad on Oct 7, 2021
This line should be in an else from the if above to make sure HeatingCoil(CoilNum).ReclaimHeatingSourceIndexNum is >= 0

Done in next commit.

Reply:
I would agree with that.